service_capacity_modeling/capacity_planner.py

```diff
@@ -503,6 +503,13 @@ def _plan_certain(
         if len(allowed_drives) == 0:
             allowed_drives.update(hardware.drives.keys())
 
+        # Get current instance type if exists
+        current_instance_name: str = extra_model_arguments.get("current_instance_name", None)
```

Comment:
let's not do this in extra args as it is generic. I'd put it into https://github.com/Netflix-Skunkworks/service-capacity-modeling/blob/main/service_capacity_modeling/interface.py#L623 as a [Clusters](https://github.com/Netflix-Skunkworks/service-capacity-modeling/blob/f0dde0d182abac85febc6cd05e5438daa59e26eb/service_capacity_modeling/interface.py#L740) object. For example

```
class Desires()
...
    CapacityPlan existing_plan
```

Then in CapacityPlan you can have the existing [requirements](https://github.com/Netflix-Skunkworks/service-capacity-modeling/blob/f0dde0d182abac85febc6cd05e5438daa59e26eb/service_capacity_modeling/interface.py#L723) expressed as cpu/mem/network.

You should be able to write a generic unit test where we give it a "wrong" cluster and then it corrects it to the "right" one.

Reply:
I did pass in the existing instance family in the interface. 

My current test case is overprovisioned (overscaled for the number of vcpus) and I've asserted it for the right one.